src/components/GestureComponents.tsx

```diff
@@ -25,16 +26,53 @@ import {
   nativeViewProps,
 } from '../handlers/NativeViewGestureHandler';
 
-export const ScrollView = createNativeWrapper<
-  PropsWithChildren<RNScrollViewProps>
->(RNScrollView, {
+import { toArray } from '../utils';
+
+export const RefreshControl = createNativeWrapper(RNRefreshControl, {
   disallowInterruption: true,
   shouldCancelWhenOutside: false,
 });
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type RefreshControl = typeof RefreshControl & RNRefreshControl;
+
+const GHScrollView = createNativeWrapper<PropsWithChildren<RNScrollViewProps>>(
+  RNScrollView,
+  {
+    disallowInterruption: true,
+    shouldCancelWhenOutside: false,
+  }
+);
+export const ScrollView = React.forwardRef<
+  RNScrollView,
+  RNScrollViewProps & NativeViewGestureHandlerProps
+>((props, ref) => {
+  const refreshControlGestureRef = React.useRef<any>();
+  const { refreshControl, waitFor, ...rest } = props;
+
+  return (
+    <GHScrollView
+      {...rest}
+      // @ts-ignore `ref` exists on `GHScrollView`
+      ref={ref}
+      waitFor={[...toArray(waitFor ? waitFor : []), refreshControlGestureRef]}
```

Comment:
What about `waitFor ?? []`?

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2137/commits/5571228feea976af1751e93422f8e8d82f30b548